src/persist-client/src/operators/shard_source.rs

```diff
@@ -63,6 +65,7 @@ pub fn shard_source<'g, K, V, T, D, F, DT, G, C>(
     client: impl Fn() -> C,
     shard_id: ShardId,
     as_of: Option<Antichain<G::Timestamp>>,
+    with_snapshot: bool,
```

Comment:
I'd like very much to avoid the bool in such a foundational public API, here and in persist_source. Mind swapping with_snapshot with a two variant enum?

Reply:
Good idea, added in a separate commit (I will rebase before I merge)